### PR TITLE
proxy support

### PR DESCRIPTION
**Pull request by murat-dogan**

### CODE REVIEW

**Comment by murat-dogan** [comment]
A note from libnice;
`Note that the proxies will only be used with TCP TURN relaying.`

**Comment by murat-dogan** [comment]
What will be the situation for libjuice? Should we check anything?

**Comment by paniksystem** [comment]
i think that libjuice does not support TURN

**Comment by paullouisageneau** [comment]
I might add TURN support to libjuice but I don't plan on adding TURN over TCP anyway to keep it light. People needing it can still use libnice instead.
Therefore you should just ignore it for libjuice, a warning is already displayed if TCP is enabled.

File: include/rtc/configuration.hpp [hunk 1]
```diff
@@ -51,8 +51,19 @@ struct IceServer {
...
+	Type type = Type::None;
+	string ip;
+	uint16_t  port;
+	string username;
+	string password;
+};
+
 struct Configuration {
 	std::vector<IceServer> iceServers;
+	ProxyServer proxyServer;
```
Comment (paullouisageneau) [inline include/rtc/configuration.hpp:66]: You should make this an optional so you don't need the `None` type anymore.

File: include/rtc/configuration.hpp [hunk 2]
```diff
@@ -51,8 +51,19 @@ struct IceServer {
 	RelayType relayType;
 };
 
+struct ProxyServer {
+	enum class Type { None = 0, Socks5, Http, Last = Http };
+
+	Type type = Type::None;
```
Comment (paullouisageneau) [inline include/rtc/configuration.hpp]: Just a suggestion, a constructor like `IceServer` could be handy.

**Comment by murat-dogan** [comment]
@paullouisageneau I deleted my fork accidentally. 
Can you merge this or should I  create it again?

**Comment by paullouisageneau** [comment]
@murat-dogan The merge went fine :+1:

**Pull request merged**